ktor-server/ktor-server-core/src/io/ktor/features/AutoOptionsResponse.kt

```diff
@@ -0,0 +1,73 @@
+package io.ktor.features
+
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.util.*
+
+/**
+ * Feature that generates an OPTIONS method handler based on the [Routing] tree.
+ *
+ * https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
+ */
+object AutoOptionsResponse : ApplicationFeature<Application, Unit, Unit> {
+    override val key = AttributeKey<Unit>("Automatic Options Response")
+
+    override fun install(pipeline: Application, configure: Unit.() -> Unit) {
+        Unit.configure()
+
+        pipeline.intercept(ApplicationCallPipeline.Call) {
+            if (call.request.local.method == HttpMethod.Options) {
```

Comment:
Why? I'm not a CORS expert so maybe I'm missing something. Any suggestion here?

Reply:
CORS is responding to `OPTIONS` as well so this will prevent it to respond properly. Have no idea how do we merge them

Reply:
Oh I see, the CORS feature. Yes that sounds like a problem.

Maybe a mechanism to mark handlers partial to be completed by others? And fallback handlers (404 and stuff) ignoring those?

Reply:
In fact you can try to intercept response pipeline and bypass request pipeline. So if there are other OPTIONS handler them you can intercept and wrap it, otherwise you can produce your own. Not sure about it but you can try this trick.